# Exploratory Analysis

Having cleaned our raw data in the previous notebook, we now have the appropriate data that we can work with.

Our main goal is to predict the `Severity` of the accident based on the different variables that we are given. Thus, we now want to find the relationship of the different variables with the `Severity` of the accident

In [1]:
import pandas as pd
import numpy as np

In [2]:
cal_accident_df = pd.read_csv('california_accident_data.csv')

We then look at the summary of the data given

In [3]:
cal_accident_df.head()

Unnamed: 0  Severity           Start_Time  Start_Lat   Start_Lng  \
0           2         2  2022-08-20 13:13:00  34.661189 -120.492822   
1           4         2  2020-12-04 01:46:00  35.395484 -118.985176   
2          13         2  2022-02-18 16:53:00  32.771645 -117.161407   
3          21         2  2020-06-15 08:17:29  34.243271 -118.424461   
4          27         2  2022-03-22 07:22:30  33.120965 -117.321490   

          Street         City         County     Zipcode  Humidity(%)  ...  \
0  Floradale Ave       Lompoc  Santa Barbara       93436         73.0  ...   
1     River Blvd  Bakersfield           Kern  93305-2649         34.0  ...   
2      Friars Rd    San Diego      San Diego       92108         40.0  ...   
3     Bartee Ave       Arleta    Los Angeles  91331-4606         64.0  ...   
4          I-5 S     Carlsbad      San Diego       92011         72.0  ...   

  Nautical_Twilight Astronomical_Twilight  Distance  Visibility  Wind_Speed  \
0               Day                   Day    0.0352        16.0        20.8   
1             Night                 Night    0.0736        16.0         0.0   
2               Day                   Day    4.5520        16.0        22.4   
3               Day                   Day    0.0000        16.0         8.0   
4               Day                   Day    2.6016        16.0         9.6   

   Temperature  Wind_Chill  Precipitation  Pressure   Road_Type  
0    20.000000   20.000000            0.0   756.666      Avenue  
1     5.555556    5.555556            0.0   756.158   Boulevard  
2    17.222222   17.222222            0.0   754.126        Road  
3    17.777778   17.777778            NaN   734.568      Avenue  
4    12.777778   12.777778            0.0   753.872  Interstate  

[5 rows x 37 columns]

In [6]:
print(cal_accident_df.shape)
print(cal_accident_df.info())

(113274, 37)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 113274 entries, 0 to 113273
Data columns (total 37 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   Unnamed: 0             113274 non-null  int64  
 1   Severity               113274 non-null  int64  
 2   Start_Time             113274 non-null  object 
 3   Start_Lat              113274 non-null  float64
 4   Start_Lng              113274 non-null  float64
 5   Street                 113113 non-null  object 
 6   City                   113274 non-null  object 
 7   County                 113274 non-null  object 
 8   Zipcode                113236 non-null  object 
 9   Humidity(%)            110220 non-null  float64
 10  Wind_Direction         110363 non-null  object 
 11  Weather_Condition      110753 non-null  object 
 12  Amenity                113274 non-null  bool   
 13  Bump                   113274 non-null  bool   
 14  Crossing               

In [8]:
cal_accident_df.describe()

Unnamed: 0       Severity      Start_Lat      Start_Lng  \
count  113274.000000  113274.000000  113274.000000  113274.000000   
mean   249718.106079       2.166075      35.634003    -119.499273   
std    144340.436636       0.407133       2.092862       1.937347   
min         2.000000       1.000000      32.543114    -124.367607   
25%    124771.500000       2.000000      33.975302    -121.518977   
50%    249713.500000       2.000000      34.237553    -118.441540   
75%    374587.500000       2.000000      37.702557    -117.921642   
max    499994.000000       4.000000      42.000100    -114.156175   

         Humidity(%)       Distance     Visibility     Wind_Speed  \
count  110220.000000  113274.000000  110743.000000  102769.000000   
mean       58.857322       0.788574      14.529272      10.440733   
std        24.220014       2.253625       4.021104       9.345152   
min         2.000000       0.000000       0.000000       0.000000   
25%        40.000000       0.000000      16.000000       4.800000   
50%        62.000000       0.040000      16.000000       9.600000   
75%        78.000000       0.694400      16.000000      14.400000   
max       100.000000      80.499200     160.000000    1316.480000   

         Temperature    Wind_Chill  Precipitation       Pressure  
count  110382.000000  80310.000000   76713.000000  110969.000000  
mean       17.684460     17.049012       0.102930     752.589038  
std         7.546929      8.092259       0.636261      19.596280  
min       -22.222222    -22.222222       0.000000     491.744000  
25%        12.777778     12.222222       0.000000     752.348000  
50%        17.222222     16.666667       0.000000     758.444000  
75%        22.222222     22.222222       0.000000     761.746000  
max        97.222222     97.222222      20.320000     976.376000

## Analysing boolean values

From the summary, we can see that columns 12 to 24 consist of boolean values. These variables are main road features such as objects, road sign, buildings. We then decided to see the relationship of the different variables with `Severity`

In [9]:
bool_cols = ['Amenity','Bump','Crossing','Give_Way','Junction','No_Exit','Railway','Roundabout','Station','Stop',
             'Traffic_Calming','Traffic_Signal','Turning_Loop']

In [20]:
# print count and corresponding ratio of T/F
for col in bool_cols:
    count = cal_accident_df[col].value_counts()
    ratio = cal_accident_df[col].value_counts(normalize=True)
    tmp = pd.concat([count, ratio], axis=1, keys=['count', 'ratio'])
    print(tmp)
    print('-'*40)

          count     ratio
Amenity                  
False    112423  0.992487
True        851  0.007513
----------------------------------------
        count    ratio
Bump                  
False  113214  0.99947
True       60  0.00053
----------------------------------------
           count     ratio
Crossing                  
False     107294  0.947208
True        5980  0.052792
----------------------------------------
           count     ratio
Give_Way                  
False     113127  0.998702
True         147  0.001298
----------------------------------------
           count     ratio
Junction                  
False     101298  0.894274
True       11976  0.105726
----------------------------------------
          count     ratio
No_Exit                  
False    113152  0.998923
True        122  0.001077
----------------------------------------
          count     ratio
Railway                  
False    112123  0.989839
True       1151  0.010161
--------------------------

From the above data, we can see that almost all of the columns have at > 90% False outcome (`Junction` has 89 % False but we round it up to 90 %)